src/components/media-player/media-player-track-group.js

```diff
@@ -244,26 +261,37 @@ function MediaPlayerTrackGroup({
         playbackRange={trackStates.mainTrack.playbackRange}
         playbackRate={playbackRate}
         posterImageUrl={posterImageUrl}
-        loadImmediately={!!trackStates.secondaryTracks.length || loadImmediately}
+        loadImmediately={isMultitrack || loadImmediately}
         onDuration={handleMainTrackDuration}
         onEndReached={handleMainTrackEndReached}
         onProgress={handleMainTrackProgress}
         onPlayStateChange={handleMainTrackPlayStateChange}
         />
-      {trackStates.secondaryTracks.map((track, index) => (
-        <MediaPlayerTrack
-          key={`${track.sourceUrl}|${index.toString()}`}
-          trackRef={trackRefs.current.secondaryTracks[index]}
-          volume={volume * track.volume}
-          sourceUrl={track.sourceUrl}
-          screenMode={MEDIA_SCREEN_MODE.none}
-          playbackRate={playbackRate}
-          loadImmediately
-          onDuration={newDuration => handleSecondaryTrackDuration(newDuration, index)}
-          onProgress={newProgress => handleSecondaryTrackProgress(newProgress, index)}
-          onPlayStateChange={newPlayState => handleSecondaryTrackPlayStateChange(newPlayState, index)}
-          />
-      ))}
+    );
+  };
+
+  const renderSecondaryTrack = (track, index) => {
+    const TrackComponent = getTrackComponent(track.sourceUrl);
+    return (
+      <TrackComponent
+        key={`${track.sourceUrl}|${index.toString()}`}
+        trackRef={trackRefs.current.secondaryTracks[index]}
+        volume={volume * track.volume}
+        sourceUrl={track.sourceUrl}
+        screenMode={MEDIA_SCREEN_MODE.none}
+        playbackRate={playbackRate}
+        loadImmediately={isMultitrack || loadImmediately}
```

Comment:
`isMultitrack` can be dropped from here to simplify the condition. It's always multitrack when a secondary track is rendered.

Reply:
Hm, that's true. So can the `loadImmediately`. Both are dropped and the prop set to `true` now.

Reply:
Yes I was just looking that the `PreloadingMediaPlayerTrack` actually throws for `loadImmediately === false` 👍